src/partest/scala/tools/partest/Util.scala

```diff
@@ -50,18 +51,24 @@ object Util {
       a.tree))
   }
 
-  def prettyArray(a: Array[_]): collection.IndexedSeq[Any] = new collection.AbstractSeq[Any] with collection.IndexedSeq[Any] {
-    def length = a.length
-
-    def apply(idx: Int): Any = a(idx) match {
-      case x: AnyRef if x.getClass.isArray => prettyArray(x.asInstanceOf[Array[_]])
-      case x => x
+  private def prettyArray0(a: Array[_]): collection.IndexedSeq[Any] = {
+    prettyArray0(mutable.ArraySeq.make(a))
+  }
+  private def prettyArray0(a: mutable.ArraySeq[_]): collection.IndexedSeq[Any] = {
+    new scala.collection.AbstractSeq[Any] with scala.collection.IndexedSeq[Any] {
+      def length = a.length
+      def apply(idx: Int): Any = a.apply(idx) match {
+        case x: Array[_] => prettyArray0(x)
+        case x => x
+      }
+      override def className = "Array"
     }
-
-    override def className = "Array"
   }
 
+  def prettyArray(a: Array[_]): collection.IndexedSeq[Any] = prettyArray0(a)
+
   implicit class ArrayDeep(val a: Array[_]) extends AnyVal {
-    def deep: collection.IndexedSeq[Any] = prettyArray(a)
+    def prettyArray: collection.IndexedSeq[Any] = prettyArray0(a)
```

Comment:
Actually, I’m thinking about something: the less code we have in this file, the more compatible with Scala.js partests are. Currently, all the tests that use the `deep` method that you renamed to `prettyArray` are blacklisted in Scala.js because this `Util.scala` file is not used by Scala.js, therefore the partests that use features defined in that file don’t compile.

Maybe it’s an opportunity to remove this `prettyArray` thing at all. Do we really need a method that wraps arbitrarily nested levels of `Arrays` to `IndexedSeq`s? I guess we usually have one or two levels of nested arrays, so the fix consists in manually calling `.toSeq` or `.map(_.toSeq).toSeq`.

Reply:
What if we keep `Array.deep` without deprecation. This method is useful to print an array or check for equality.

Reply:
I’m not sure. To print an array I find it simple to call `.toSeq`. To check for equality we can use Java’s `Arrays.deepEquals`. What do other people think?

Reply:
```scala
Array(Array(1)).toSeq.toString // WrappedArray([I@6fc1a8f6)
```

Reply:
If we keep it, `deep` is certainly a not a good name. Are there any other use cases than `toString`? If not, there should instead be a method that returns the string representation directly.

Personally, I think the method is not widely useful enough for the standard libarary.

Reply:
It's for pretty-printing and checking for equality in tests. If you use Arrays.deepEquals, you only have one.

Reply:
The very fact that `deep()` returns an `IndexedSeq[Any]` is a huge smell. It's basically not type parametric. If I manipulate an `Array[T]` and I call `deep()` on it, I am at the mercy of some of the `T`s being `Array`s. It's the same huge problem that JavaScript `Promise`s face.

Kill it.